mods-unpacked/ITR-SmarterDealer/bruteforce.gd

```diff
@@ -491,113 +492,22 @@ static func GetBestChoiceAndDamage_Internal(roundType: int, liveCount: int, blan
 	if cache.has(ahash) and not isTopLayer:
 		return cache[ahash].clone()
 
-	# Double or nothing round has special logic
-	var donLogic = roundType == ROUNDTYPE_DOUBLEORNOTHING and player.player_index == 1
-
 	if liveCount == 0 and blankCount == 0:
-		var playerItemscore := player.sum_items()
-
-		var smokeAmount: int = min(player.cigarettes, player.max_health - player.health)
-		if player.player_index != 0:
-			smokeAmount = 0
-
-		var opponentSmokeAmount := 0
-		if opponent.player_index == 0:
-			opponentSmokeAmount = min(opponent.cigarettes, opponent.max_health - opponent.health)
-
-		var opponentItemscore := opponent.sum_items()
+		var deathChance: Array[float] = [0.0, 0.0]
 
 		var health: Array[float] = [0.0, 0.0]
-		health[player.player_index] = player.health + smokeAmount
-		health[opponent.player_index] = opponent.health + opponentSmokeAmount
+		health[player.player_index] = player.health
+		health[opponent.player_index] = opponent.health
```

Comment:
What's the thought behind removing this? Player 0 can regain health on the beginning of their turn, so it's sometimes beneficial to let the cigarettes take up space in the inventory so the player can't draw more items.

Reply:
Good catch. I added back the logic for smoking at the start of the next round.

Reply:
Hmm... this sometimes leads to weird behavior. The dealer won't use an inverter on the last blank shell in the chamber when you have cigarettes because he views it as trading his high-value inverter for your low-value cigarettes. Maybe that's more of a symptom of undervaluing cigarettes than anything else, though. (Without the smokeAmount logic the dealer would go out of his way to make sure you weren't allowed to smoke before the end of the round, but then you could just smoke at the beginning of the next round.)